### ISSUE TITLE
howdy does not respect $EDITOR

### ISSUE THREAD

**Issue by NuclearDog** [boltgolt/howdy#333]
howdy (2.5.1) does not respect the standard unix $EDITOR environment variable.

In `src/cli/config.py`, should replace:

```
# Default to the nano editor
editor = "/bin/nano"
```

With something more like:
```
editor = os.environ.get("EDITOR", "nano")
```

(That's also switched from using `/bin/nano` -> `nano`. This will respect the user's $PATH variable as far as locating the nano binary.)

**Comment by boltgolt** [comment]
Hmm, but howdy does fetch that exact variable a few lines below that? It only uses nano if no `/etc/alternatives/editor` or `$EDITOR` is found

**Comment by NuclearDog** [comment]
Ah sorry about that. I saw EDITOR wasn't doing anything, saw the "editor = '/bin/nano'" line, and just kinda stopped reading there. Will you accept lack of coffee as an excuse? :disappointed: 

Looks like my issue is more the prioritization -- I'd argue that preferring `/etc/alternatives/editor` over `$EDITOR` is backwards and that `if`/`elif` should be reversed. `/etc/alternatives/editor` is set on the system level by the administrator, while `$EDITOR` is supposed to be the per-user override.

The current behaviour is definitely opposite any other application I've run into, as apparently my `/etc/alternatives/editor` is linked to nano and this is the first time I've noticed because I have `$EDITOR='vim'` set and that's been respected by e.g., visudo, etc.

**Comment by boltgolt** [comment]
Hm, okay. Fixing in next release

**Comment by boltgolt** [comment]
Implemented exactly as suggested

**Comment by xz-dev** [comment]
The bug is back.

![图片](https://user-images.githubusercontent.com/32761048/90757769-a1b1fe00-e310-11ea-933d-c72d658ccaf1.png)

**Comment by arifer612** [comment]
@xz-dev The problem at hand seems to be due to the fact that you are elevated when running `sudo howdy config`. Your user-defined `EDITOR` is no longer relevant. A temporary fix you can do to handle this would be to change the symlink for your root default editor to `nvim`.

    sudo ln -sf /etc/alternatives/editor /path/to/nvim

@boltgolt I think the closure of this issue should be reconsidered.

**Comment by boltgolt** [comment]
@arifer612 what would a better solution be in your opinion? I think it's hard to make this work everywhere as everyone expects

**Comment by arifer612** [comment]
@boltgolt I apologise for sounding very direct, I did not mean for that.

I was searching for a way to do so and I think I found a method that might work for this use-case. Hooking the `$user` name into the following:

    editor = subprocess.run(["su", "-", "$user", "-c" "source ~/.profile; echo $EDITOR"], stdout=subprocess.PIPE).stdout.decode()

It gave me the right `$EDITOR` when executed as root. I have yet to see if it can be fit `config.py` without any issues yet, but I can look into with your permission.
